components/usage/pkg/stripe/stripe.go

```diff
@@ -122,6 +130,38 @@ func (c *Client) updateUsageForCustomer(ctx context.Context, customer *stripe.Cu
 	}, nil
 }
 
+// GetUpcomingInvoice fetches the upcoming invoice for the given team
+func (c *Client) GetUpcomingInvoice(ctx context.Context, teamId string) (*StripeInvoice, error) {
+	query := fmt.Sprintf("metadata['teamId']:'%s'", teamId)
+	searchParams := &stripe.CustomerSearchParams{
+		SearchParams: stripe.SearchParams{
+			Query:   query,
+			Expand:  []*string{stripe.String("data.subscriptions")},
+			Context: ctx,
+		},
+	}
+	iter := c.sc.Customers.Search(searchParams)
+	if !iter.Next() {
+		return nil, fmt.Errorf("failed to find customer data for team %s", teamId)
```

Comment:
Worth returning a special error for this case. We want the API to return a useful error message to the client. In this case, we want to say that for a given TeamID, there's no customer in Stripe. The general error here can't be classified (or compared, aside from string comparisons).

The pattern for this is to do:

```
var CustomerNotFoundError = errors.New(...)
...

if errors.Is(err, CustomerNotFoundError) {
  return NotFound
}
```